src/test/java/com/puppycrawl/tools/checkstyle/meta/XmlMetaWriter.java

```diff
@@ -0,0 +1,159 @@
+            final Element properties = doc.createElement("properties");
+            checkModule.appendChild(properties);
+            for (ModulePropertyDetails modulePropertyDetails : moduleDetails.getProperties()) {
+                final Element property = doc.createElement("property");
+                properties.appendChild(property);
+                property.setAttribute("name", modulePropertyDetails.getName());
+                property.setAttribute("type", modulePropertyDetails.getType());
+                if (modulePropertyDetails.getDefaultValue() != null) {
+                    property.setAttribute("default-value",
+                            modulePropertyDetails.getDefaultValue());
+                }
+                if (modulePropertyDetails.getValidationType() != null) {
+                    property.setAttribute("validation-type",
+                            modulePropertyDetails.getValidationType());
+                }
+                final Element propertyDesc = doc.createElement("description");
+                propertyDesc.appendChild(doc.createCDATASection(
+                        modulePropertyDetails.getDescription()));
+                property.appendChild(propertyDesc);
+            }
+        }
+    }
+
+    private static void writeToFile(Document document, ModuleDetails moduleDetails)
+            throws TransformerException {
+        final String rootOutputPath = System.getProperty("user.dir") + "/src/main/resources";
+        String fileSeperator = System.getProperty("file.separator");
+        if (System.getProperty("os.name").toLowerCase(Locale.ENGLISH).contains("win")) {
+            fileSeperator = "\\" + fileSeperator;
+        }
+        final String moduleFilePath = FILEPATH_CONVERSION
+                .matcher(moduleDetails.getFullQualifiedName())
+                .replaceAll(fileSeperator);
+        final String modifiedPath;
+        if (moduleFilePath.contains("puppycrawl")) {
+            final int idxOfCheckstyle =
+                    moduleFilePath.indexOf("checkstyle") + "checkstyle".length();
+            // make sure all folders are created
+            modifiedPath = rootOutputPath + System.getProperty("file.separator")
+                    + moduleFilePath.substring(0, idxOfCheckstyle)
+                    + System.getProperty("file.separator") + "meta"
+                    + System.getProperty("file.separator")
+                    + moduleFilePath.substring(idxOfCheckstyle + 1) + ".xml";
+        }
+        else {
+            String moduleName = moduleDetails.getName();
+            if (moduleDetails.getModuleType() == ModuleType.CHECK) {
+                moduleName += "Check";
+            }
+            modifiedPath = rootOutputPath + System.getProperty("file.separator") + "checkstylemeta-"
+                    + moduleName + ".xml";
+        }
+        if (!moduleDetails.getDescription().isEmpty()) {
+            // log to ensure folders exist
```

Comment:
`log to`
please rephase comment, you do not any logging in this block.

Reply:
I have removed, since its solved.